pep-0xxx.rst

```diff
@@ -0,0 +1,1250 @@
+
+        @abstractmethod
+        def negotiated_protocol(self) -> Optional[Union[NextProtocol, bytes]]:
+            """
+            Returns the protocol that was selected during the TLS handshake.
+            This selection may have been made using ALPN, NPN, or some future
+            negotiation mechanism.
+
+            If the negotiated protocol is one of the protocols defined in the
+            ``NextProtocol`` enum, the value from that enum will be returned.
+            Otherwise, the raw bytestring of the negotiated protocol will be
+            returned.
+
+            If ``Context.set_inner_protocols()`` was not called, if the other
+            party does not support protocol negotiation, if this socket does
+            not support any of the peer's proposed protocols, or if the
+            handshake has not happened yet, ``None`` is returned.
+            """
+
+        @property
+        @abstractmethod
+        def context(self) -> Context:
+            """
+            The ``Context`` object this socket is tied to.
+            """
+
+        @abstractproperty
+        def negotiated_tls_version(self) -> Optional[TLSVersion]:
+            """
+            The version of TLS that has been negotiated on this connection.
+            """
+
+        @abstractmethod
+        def unwrap(self) -> socket.socket:
+            """
+            Cleanly terminate the TLS connection on this wrapped socket. Once
+            called, this ``TLSWrappedSocket`` can no longer be used to transmit
+            data. Returns the socket that was wrapped with TLS.
+            """
+
+
+Buffer
+~~~~~~
+
+The buffer-wrapper ABC will be defined by the ``TLSWrappedBuffer`` ABC, which
+has the following definition::
+
+    class TLSWrappedBuffer(metaclass=ABCMeta):
+        @abstractmethod
+        def read(self, amt=None: int) -> bytes:
+            """
+            Read up to ``amt`` bytes of data from the input buffer and return
+            the result as a ``bytes`` instance. If ``amt`` is ``None``, will
+            attempt to read until either EOF is reached or until further
+            attempts to read would raise either ``WantReadError`` or
+            ``WantWriteError``.
+
+            Raise ``WantReadError`` or ``WantWriteError`` if there is
+            insufficient data in either the input or output buffer and the
+            operation would have caused data to be written or read.
+
+            As at any time a re-negotiation is possible, a call to ``read()``
+            can also cause write operations.
+            """
+
+        @abstractmethod
+        def readinto(self, buffer: Any, amt=None: int) -> int:
+            """
+            Read up to ``amt`` bytes of data from the input buffer into
+            ``buffer``, which must be an object that implements the buffer
+            protocol. Returns the number of bytes read. If ``amt`` is ``None``,
+            will attempt to read until either EOF is reached or until further
```

Comment:
This comment defines three conditions when this read will stop:

1. When EOF is reached
2. When further attempts to read would raise `WantWriteError` or `WantReadError`
3. When the buffer is full.

Would it be clearer to amend this docstring to use that numbered-list form, instead of the current long sentence?